src/View/Sidebar.vala

```diff
@@ -1611,9 +1613,27 @@ namespace Marlin.Places {
             bool show_properties = show_mount || show_unmount || show_eject || uri == Marlin.ROOT_FS_URI;
 
             if (is_plugin) {
+                MenuModel model;
+                string action_group_namespace;
+                ActionGroup action_group;
+
+                store.get (iter, Column.MENU_MODEL, out model);
+                store.get (iter, Column.ACTION_GROUP_NAMESPACE, out action_group_namespace);
+                store.get (iter, Column.ACTION_GROUP, out action_group);
+
                 var menu = new PopupMenuBuilder ()
                 .add_open (open_shortcut_cb);
-                menu.build ().popup_at_pointer (event);
+                if (model == null) {
+                    menu.build ().popup_at_pointer (event);
+                } else {
+                    menu
+                    .add_separator ()
+                    .add_open_tab (open_shortcut_in_new_tab_cb)
+                    .add_open_window (open_shortcut_in_new_window_cb)
+                    .add_separator ()
+                    .build_from_model (model, action_group_namespace, action_group)
+                    .popup_at_pointer (event);
```

Comment:
We usually don't use this syntax, but it's fine here. I would indent the lines after menu to be clear where the call chain ends.